sst/SimEngCoreWrapper.cc

```diff
@@ -0,0 +1,368 @@
+    bool escaped = escapeDouble || escapeSingle;
+    char currChar = trimmedStrArgs.at(x);
+    if (captureEscape) {
+      captureEscape = false;
+      str += currChar;
+    }
+    // This if statement check for an escaped '\' in the string.
+    // Any character after the '\' is appended to the current argument,
+    // without any delimiting or escape behaviour.
+    else if (currChar == '\\') {
+      captureEscape = true;
+    } else if (escaped) {
+      // If a portion of the argument string starts with a single quote (") and
+      // we encounter another single quote, capture the substring enclosed by a
+      // valid set of single quotes into an argument without producing any
+      // delimiting or escape behavior even with double quotes.
+      // e.g "arg1=1 arg2='"Hi"' arg3=2" will be parsed as
+      // std::vector<std::string>{arg1=1, arg2="Hi", arg3=2}
+      if (currChar == '\'' && escapeSingle) {
+        escapeSingle = 0;
+      }
+      // If a portion of the argument string starts with a double quote (") and
+      // we encounter another double quote, capture the substring enclosed by a
+      // valid set of double quotes into an argument without producing any
+      // delimiting or escape behavior even with single quotes.
+      // e.g "arg1=1 arg2="James' Car" arg3=2" will be parsed as
+      // std::vector<std::string>{arg1=1, arg2=James' Car, arg3=2}
+      else if (currChar == '\"' && escapeDouble) {
+        escapeDouble = 0;
+      } else {
+        str += currChar;
+      }
+    } else {
+      if (currChar == ' ') {
+        if (str != "") {
+          args.push_back(str);
+          str = "";
+        }
+      }
+      // Check for escape character ("), this signals the algorithm to capture
+      // any char inside a set of ("") without producing any delimiting or
+      // escape behavior.
+      else if (currChar == '\"') {
+        escapeDouble = 1;
+        // Check for escape character ('), this signals the algorithm to capture
+        // any char inside a set of ('') without producing any delimiting or
+        // escape behavior.
+      } else if (currChar == '\'') {
+        escapeSingle = 1;
+      } else {
+        str += currChar;
+      }
+    }
+  }
+  if (escapeSingle || escapeDouble) {
+    std::string err;
+    output_.verbose(CALL_INFO, 1, 0, R"(
+           Parsing failed: Invalid format - Please make sure all
+           characters/strings are escaped properly within a set single or 
+           double quotes. To escape quotes use (\\\) instead of (\).\n
+           )");
+    std::cerr << "Error occured at index " << index
+              << " of the argument string - substring: "
+              << "[ " << str << " ]" << std::endl;
+    std::exit(EXIT_FAILURE);
+  }
+  args.push_back(str);
+  return args;
+}
+
+void SimEngCoreWrapper::fabricateSimEngCore() {
+  output_.verbose(CALL_INFO, 1, 0, "Setting up SimEng Core\n");
+  if (simengConfigPath_ != "") {
+#ifdef SIMENG_ENABLE_SST_TESTS
```

Comment:
Again this needs to be done because of the injectable LLVM dependency. See [comment](https://github.com/UoB-HPC/SimEng/pull/271#discussion_r1016632653)